From cbe9fe3834d08758bc787ac0800afb9ece6ed782 Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Tue, 12 Nov 2013 17:10:04 +0100
Subject: [PATCH 01/14] pc: drop virtio-balloon-pci event_idx compat property

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <20131112121004.12eb9431@redhat.com>
Patchwork-id: 55664
O-Subject: [RHEL-7.0 qemu-kvm PATCH] pc: drop virtio-balloon-pci event_idx compat property
Bugzilla: 1029539
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>

Bugzilla: 1029539
Brew-build: https://brewweb.devel.redhat.com/taskinfo?taskID=6564546

Due to a bug, RHEL6 has never disabled the virtio-balloon-pci
driver's event_idx property for machine types rhel6.0.0 and
rhel6.1.0. The end result is that migration from RHEL6 to RHEL7
fails for those machine types.

The most obvious way to fix this problem is to fix RHEL6 to
really disable the property. But this won't solve the problem
for old RHEL6 binaries. Actually, it will make migration
incompatible between old RHEL6 binaries and newer ones.

This commit makes the mistake official instead by dropping
the compat property for RHEL7. This fixes migration between
RHEL6 and RHEL7 without adding any new problems.

Signed-off-by: Luiz Capitulino <lcapitulino@gmail.com>
---
 hw/i386/pc_piix.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

PS: Will post RHEL6.6 patch later.
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/pc_piix.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 8bae4c5..e6771a9 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1030,6 +1030,11 @@ static QEMUMachine pc_machine_rhel620 = {
     DEFAULT_MACHINE_OPTIONS,
 };
 
+/* 
+ * NOTE: We don't have the event_idx compat entry for the
+ * virtio-balloon-pci driver because RHEL6 doesn't disable
+ * it either due to a bug (see RHBZ 1029539 fo more info)
+ */
 #define PC_RHEL6_1_COMPAT \
     PC_RHEL6_2_COMPAT,\
     {\
@@ -1061,10 +1066,6 @@ static QEMUMachine pc_machine_rhel620 = {
         .property = "event_idx",\
         .value    = "off",\
     },{\
-        .driver   = "virtio-balloon-pci",\
-        .property = "event_idx",\
-        .value    = "off",\
-    },{\
         .driver   = "usb-kbd",\
         .property = "serial",\
         .value    = "1",\
-- 
1.7.1

